sw/device/lib/testing/test_framework/check.h

```diff
@@ -25,14 +25,15 @@
 
 /**
  * Checks that the given condition is true. If the condition is false, this
- * function logs and then aborts.
+ * function logs and then returns a `kInternal` error.
  *
  * @param condition An expression to check.
  * @param ... Arguments to a LOG_* macro, which are evaluated if the check
  * fails.
  */
-#define CHECK(condition, ...)                        \
-  do {                                               \
+#define CHECK_IMPL(condition, ...)                   \
```

Comment:
I pulled down you PR to see why it was failing CI, and I think you want this to be:
```c
#define CHECK_IMPL(condition, ...)                   \
  ({                                                 \
    if (!(condition)) {                              \
      /* NOTE: because the condition in this if      \
         statement can be statically determined,     \
         only one of the below string constants      \
         will be included in the final binary.*/     \
      if (OT_VA_ARGS_COUNT(_, ##__VA_ARGS__) == 0) { \
        LOG_ERROR("CHECK-fail: " #condition);        \
      } else {                                       \
        LOG_ERROR("CHECK-fail: " __VA_ARGS__);       \
      }                                              \
      INTERNAL();                                    \
    }                                                \
    OK_STATUS();                                     \
  })
```

Reply:
Many Thanks Tim! Out of curiosity, could you reproduce locally the failure?

Reply:
Yes, I was able to reproduce it locally on my FPGA.

Reply:
Doesn't look like your latest push includes the above suggestion.